inspirehep/utils/url.py

```diff
@@ -58,7 +58,10 @@ def is_pdf_link(url):
     except requests.exceptions.RequestException:
         return False
 
-    magic_number = next(response.iter_lines(1))
+    try:
+        magic_number = next(response.iter_lines(1))
+    except StopIteration:
+        return False
```

Comment:
Nope, you cannot use `response.raw.read(4)` because you don't know in advance how many gzipped bytes you will have to read before decoding enough to have the first 4 bytes.

So I just added a test for your change and we're good.